samplesheets/views_api.py

```diff
@@ -137,6 +147,317 @@ def post(self, request, *args, **kwargs):
+            sodar_uuid=self.kwargs.get('project')
+        ).first()
+        irods_requests = IrodsDataRequest.objects.filter(project=project)
+
+        # For superusers, owners and delegates,
+        # display active/failed requests from all users
+        if (
+            self.request.user.is_superuser
+            or project.is_delegate(self.request.user)
+            or project.is_owner(self.request.user)
+        ):
+            requests = irods_requests.filter(status__in=['ACTIVE', 'FAILED'])
+        else:
+            # For regular users, dispaly their own requests regardless of status
+            requests = irods_requests.filter(user=self.request.user)
+
+        if not requests:
+            return Response(
+                {
+                    'detail': 'No iRODS data requests found',
+                },
+                status=status.HTTP_200_OK,
+            )
+
+        return Response(
+            {
+                'detail': 'iRODS data requests listed',
+                'requests': requests,
+            },
+            status=status.HTTP_200_OK,
+        )
+
+
+class IrodsRequestCreateAPIView(
+    IrodsRequestModifyMixin, SODARAPIBaseProjectMixin, CreateAPIView
+):
+    """
+    Create an iRODS data request for a project.
+
+    **URL:** ``/samplesheets/api/irods/request/create/{Project.sodar_uuid}``
+
+    **Methods:** ``POST``
+    """
+
+    http_method_names = ['post']
+    permission_required = 'samplesheets.edit_sheet'
+    serializer_class = IrodsRequestSerializer
+
+    def post(self, request, *args, **kwargs):
+        """POST request for creating an iRODS data request"""
+        project = self.get_project()
+        serializer = self.get_serializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save(
+                project=project,
+                user=self.request.user,
+            )
+        else:
+            raise ValidationError('Invalid data: {}'.format(serializer.errors))
+
+        irods_request = IrodsDataRequest.objects.filter(
+            sodar_uuid=serializer.data.get('sodar_uuid')
+        ).first()
+        # Create timeline event
+        self.add_tl_create(irods_request)
+        # Add app alerts to owners/delegates
+        self.add_alerts_create(project)
+
+        return Response(
+            {
+                'detail': 'iRODS data request created',
+                'request': irods_request.sodar_uuid,
+            },
+            status=status.HTTP_200_OK,
+        )
+
+
+class IrodsRequestUpdateAPIView(
+    IrodsRequestModifyMixin, SODARAPIBaseProjectMixin, UpdateAPIView
+):
+    """
+    Update an iRODS data request for a project.
+
+    **URL:** ``/samplesheets/api/irods/request/update/{IrodsDataRequest.sodar_uuid}``
+
+    **Methods:** ``POST``
+    """
+
+    http_method_names = ['post']
+    # permission_required = 'samplesheets.edit_sheet'
+    permission_classes = (IsAuthenticated,)
+
+    def check_irods_permissions(self, request):
```

Comment:
This method seems to be duplicated in `IrodsRequestDeleteAPIView` and in both cases it's only called once within the view. I suggest to remove the repetition by making this a mixin.

The docstring is a bit confusing, as this is not an override of anything and the API views should have nothing to do with CSRF tokens. Maybe a forgotten copypaste from somewhere? :)